gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,541 @@
+    if (!Strings.isNullOrEmpty(template)) {
+      insert.setSourceInstanceTemplate(template);
+    }
+    return insert.execute();
+  }
+
+  public Operation terminateInstance(String projectId, String zone, String InstanceId)
+      throws IOException {
+    zone = nameFromSelfLink(zone);
+    return compute.instances().delete(projectId, zone, InstanceId).execute();
+  }
+
+  public Operation terminateInstanceWithStatus(
+      String projectId, String zone, String instanceId, String desiredStatus)
+      throws IOException, InterruptedException {
+    zone = nameFromSelfLink(zone);
+    Instance i = getInstance(projectId, zone, instanceId);
+    if (i.getStatus().equals(desiredStatus)) {
+      return compute.instances().delete(projectId, zone, instanceId).execute();
+    }
+    return null;
+  }
+
+  public Instance getInstance(String projectId, String zone, String instanceId) throws IOException {
+    zone = nameFromSelfLink(zone);
+    return compute.instances().get(projectId, zone, instanceId).execute();
+  }
+
+  /**
+   * Return all instances that contain the given labels
+   *
+   * @param projectId
+   * @param labels
+   * @return
+   * @throws IOException
+   */
+  public List<Instance> getInstancesWithLabel(String projectId, Map<String, String> labels)
+      throws IOException {
+    Compute.Instances.AggregatedList request = compute.instances().aggregatedList(projectId);
+    request.setFilter(buildLabelsFilterString(labels));
+    Map<String, InstancesScopedList> result = request.execute().getItems();
+    List<Instance> instances = new ArrayList<>();
+    for (InstancesScopedList instancesInZone : result.values()) {
+      if (instancesInZone.getInstances() != null) {
+        instances.addAll(instancesInZone.getInstances());
+      }
+    }
+    return instances;
+  }
+
+  public InstanceTemplate getTemplate(String projectId, String templateName) throws IOException {
+    return compute.instanceTemplates().get(projectId, templateName).execute();
+  }
+
+  public void insertTemplate(String projectId, InstanceTemplate instanceTemplate)
+      throws IOException {
+    compute.instanceTemplates().insert(projectId, instanceTemplate).execute();
+  }
+
+  public void deleteTemplate(String projectId, String templateName) throws IOException {
+    compute.instanceTemplates().delete(projectId, templateName).execute();
+  }
+
+  public List<InstanceTemplate> getTemplates(String projectId) throws IOException {
+    List<InstanceTemplate> instanceTemplates =
+        compute.instanceTemplates().list(projectId).execute().getItems();
+    if (instanceTemplates == null) {
+      instanceTemplates = Collections.emptyList();
+    }
+
+    // Sort by name
+    instanceTemplates.sort(Comparator.comparing(InstanceTemplate::getName));
+
+    return instanceTemplates;
+  }
+
+  /**
+   * Creates persistent disk snapshot for Compute Engine instance. This method blocks until the
```

Comment:
Prefer incorporating behavior like "blocking/non-blocking" into the method name itself. Downstream callers are more likely to pick-up on that rather than a note in the javadoc alone. Suggest using: "createSnapshotSync" or something similar.

Reply:
Whichever convention you choose, should also be applied consistently throughout.

Reply:
Done, although does this apply to waitForOperationCompletion? There the naming already implies that it blocks.